drivers/src/nrfx_nvmc.c

```diff
@@ -415,6 +457,18 @@ void nrfx_nvmc_words_write(uint32_t addr, void const * src, uint32_t num_words)
     nvmc_readonly_mode_set();
 }
 
+uint16_t nrfx_nvmc_otp_halfword_read(uint32_t addr)
```

Comment:
I'd really prefer `addr` over `otp_index` as argument, since my use case deals with addresses, not indices. `addr` is arguably more useful since it's easier to translate from `otp_index` to `addr` than the other way.

Also, I think the halfword_write is useful elsewhere than just OTP since all flash has limits on the number of writes to a given word.

Reply:
I left the write functions as they were, but encased the read in #ifdefs like you suggested. Please take a look.

Reply:
Checked, LGTM, thanks!